### PR TITLE
Random number generation seeding structure refactor

### PR DESCRIPTION
**Pull request by rnugent3**

### CODE REVIEW

**Comment by rnugent3** [comment]
Checking build and tests on GitHub because ACE-IT told me no. Please hold off review.

File: HEC.FDA.Model/metrics/SystemPerformanceResults.cs
Comment (Brennan1994) [inline HEC.FDA.Model/metrics/SystemPerformanceResults.cs:21]: This seems impactful for plotting. We were plotting using a different histogram for some reason. Maybe that's why our plot was all stair steppy

File: HEC.FDA.Model/paireddata/IPairedDataProducer.cs
Comment (Brennan1994) [inline HEC.FDA.Model/paireddata/IPairedDataProducer.cs:5]: Do we need to keep this compute is deterministic?

File: HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs
Comment (Brennan1994) [inline HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs:349]: Want to talk about whether we still need this bool.

File: HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs
Comment (Brennan1994) [inline HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs:638]: Looks like we previously wanted the mean here. now we're getting the iteration 1 sample. Are we still good?

File: HEC.FDA.Model/compute/ImpactAreaScenarioSimulation.cs
Comment (Brennan1994) [inline HEC.FDA.Model/compute/ImpactAreaScenarioSimulation.cs:137]: I'm interested in whether it makes sense to have each individual object track their seeds, or for ImpactAreaScenarioSimulation to track them for them and inject the right one.  What were your thoughts on that?

**Review by Brennan1994 — CHANGES_REQUESTED** [review]

**Comment by Brennan1994** [comment]
There's more here than i can chew on. Lets chat about a few things in your office today, and we'll approve or identify changes after that

File: HEC.FDA.Model/metrics/SystemPerformanceResults.cs
Comment (rnugent3) [reply HEC.FDA.Model/metrics/SystemPerformanceResults.cs:21]: resolved in person

File: HEC.FDA.Model/paireddata/IPairedDataProducer.cs
Comment (rnugent3) [reply HEC.FDA.Model/paireddata/IPairedDataProducer.cs:5]: resolved

File: HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs
Comment (rnugent3) [reply HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs:638]: Add comments to note that iteration is ignored for compute is deterministic

File: HEC.FDA.Model/compute/ImpactAreaScenarioSimulation.cs
Comment (rnugent3) [reply HEC.FDA.Model/compute/ImpactAreaScenarioSimulation.cs:137]: resolved

File: HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs
Comment (rnugent3) [reply HEC.FDA.Model/stageDamage/ImpactAreaStageDamage.cs:638]: I have added public XML comments on all of the objects that can be sampled

**Review by Brennan1994 — APPROVED** [review]

**Comment by Brennan1994** [comment]
The suspected result changing modification here is moving from a sampling schema which samples a small number of samples from a large number of random number generators to a new schema of a single random number generator with many samples. 

The results appear better distributed after the changes in this request. The design choice between these two sampling schemes was presented in a code review with computer scientists outside the HEC-FDA team, and checked against industry best practices as documented by Microsoft. Generating many values from a single generator is the proper use of _Random.cs_. 

"Both to improve performance and to avoid inadvertently creating separate random number generators that generate identical numeric sequences, we recommend that you create one Random object to generate many random numbers over time, instead of creating new Random objects to generate one random number." - Microsoft docs
https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-random

**Pull request merged**
